src/Api.elm

```diff
@@ -0,0 +1,96 @@
+module Api exposing (..)
+
+import Decoders
+import Dict exposing (Dict)
+import Encoders
+import Http
+import Http.Detailed exposing (Error(..))
+import Json.Decode exposing (Decoder, decodeString)
+import Model exposing (..)
+
+
+fetchUsers : Cmd Msg
+fetchUsers =
+    Http.get
+        { url = "https://configure"
+        , expect =
+            Http.expectStringResponse
+                GetUsersResponse
+                (responseToPayload Dict.empty Decoders.users)
+        }
+
+
+createUser : User -> Cmd Msg
+createUser user =
+    Http.post
+        { url = "https://configure"
+        , expect =
+            Http.expectStringResponse
+                UpsertUserResponse
+                (responseToPayload (Dict.fromList [ ( 422, Decoders.userUpsertErrors ) ]) Decoders.userResponse)
+        , body = Http.jsonBody (Encoders.user user)
+        }
```

Comment:
I find myself navigating more confidently in the horizontal organisation, because it speaks more about the code than about domain objects. The code deals with domain objects - I deal with the code. I might be biased though, as most of the repos I'we worked with didn't have more than 4-5 types of domain objects. But then again, another advantage of the horizontal organisation is that it's easier to copy relevant bits of code from surrounding functions.